macros/src/codegen.rs

```diff
@@ -454,6 +476,12 @@ pub fn generate_code(sm: &ParsedStateMachine) -> proc_macro2::TokenStream {
         quote! {}
     };
 
+    let (is_async, is_async_trait) = if sm_is_async {
+        (quote! { async }, quote! { #[async_trait::async_trait] })
```

Comment:
I _think_ this may need to be `smlang::async_trait::async_trait`. There's no guarantee that a user crate will have `async_trait` in scope

Reply:
The user will have to use `async_trait` for impl anyway. I'm not sure if it's best to switch `async_trait` from dev-dep to dep here and export it so end-user can also use it, like that: https://github.com/yaak-ai/smlang-rs/commit/6a52778d238018ccca0460bb1e8b89b500267465; or let the user manage and `use` its own `async_trait` - that might be more flexible, but also might looks a bit ugly.
WDYT?

Reply:
I'd prefer we re-export it. AFAIK they need to be the same version anyways